planet/src/main/java/com/google/planet/data/ItineraryGenerator.java

```diff
@@ -47,74 +47,56 @@ public Itinerary generateItinerary(List<Event> events) {
-        int numOfNodes = events.size();
-        Duration[][] travelTimeGraph = new Duration[numOfNodes][numOfNodes];
-
-        String GoogleApiKey = "AIzaSyDK36gDoYgOj4AlbCqh1IuaUuTlcpKF0ns";
-        String[] origins = getListOfAddresses(events);
-        String[] destinations = getListOfAddresses(events);
-        GeoApiContext context = new GeoApiContext.Builder(new GaeRequestHandler.Builder())
-                            .apiKey(GoogleApiKey)
-                            .build();
-       try {
-            DistanceMatrix distanceMatrix = DistanceMatrixApi.getDistanceMatrix(context,
-                                                         origins,
-                                                         destinations).await();
+    // Function that creates an itinerary by scheduling each event in order.
+    private List<ItineraryItem> scheduleItineraryInOrder(List<Event> events, int openingTime, int endingTime) 
+                                    throws ItineraryException{
+        List<ItineraryItem> items = new ArrayList();
+        DirectionsRoute directionsRoute = getDirectionsRoute(events, ItineraryOrder.UNOPTIMIZED);
+        DirectionsLeg[] directionsLegs = directionsRoute.legs;
+        int start = openingTime; 
+        for (int i = 0; i < events.size(); i++){
+            Event event = events.get(i);
+            // Add the event as an itinerary item if the remaining available time is longer than the
+            // event duration.
+            if (start + event.getDurationInMinutes() <= endingTime) {
+                ItineraryItem item = new ItineraryItem(event.getName(), event.getAddress(),
+                    TimeRange.fromStartDuration(start, event.getDurationInMinutes()));
+                items.add(item);
 
-            for (int i = 0; i < distanceMatrix.rows.length; i++) {
-                DistanceMatrixRow currentRow = distanceMatrix.rows[i];
-                for (int j = 0; j < currentRow.elements.length; j++) {
-                    DistanceMatrixElement currentElement = currentRow.elements[j];
-                    if (currentElement.duration != null) {
-                        travelTimeGraph[i][j] = currentElement.duration;
-                    } else {
-                        errorMessage = "Oops, one of your addresses is invalid, please use a valid address.";
-                    }
-                }
+                // Update the next event's start time
+                start += event.getDurationInMinutes() + getTravelDurationInMinutes(directionsLegs[i].duration);
+            }else {
+                throw new ItineraryException("Sorry, you have too many events in a day! Try removing some events or shorten their duration");
             }
-            return travelTimeGraph;
-        } catch (Exception e) {
-            errorMessage = "Oops, one of your addresses is invalid, please use a valid address.";
-            return travelTimeGraph;
         }
+        return items;
```

Comment:
Check the lines 93 -> 114 of the same file. It's almost the same code with a small difference in getting the event's index.

Reply:
You're right. I guess my thinking was whether this will make the code less readable. Since the waypointOrder will be null if the route is not optimized, we need to add more if statements to assign the right eventIndex, which could make it hard to read for people unfamiliar with the code. What do you think @nabil-dbz @EkkiD

Reply:
If it is indeed a null object, I would consider working around it by writing a small function that gets the waypoints:

`getWaypoints(DirectionsRoute)` which returns the waypoints object if it exists, or constructs  waypoints object with [1,2,...n]

Reply:
When I manually make a request (http://maps.googleapis.com/maps/api/directions/outputFormat?parameters) it seems like the waypointOrder is empty...

Reply:
Actually when I tested it with the java client library it does return the correct order!
I just changed the code

Reply:
Sorry for the confusion here